internal/libyaml/api_test.go

```diff
@@ -0,0 +1,527 @@
+	// Handle boolean values
+	if name == "true" {
+		return 1
+	}
+	if name == "false" {
+		return 0
+	}
+
+	// Try parsing as int first
+	if val, err := strconv.Atoi(name); err == nil {
+		return val
+	}
+
+	// Use IntOrStr to parse other constants
+	ios := IntOrStr{}
+	err := ios.FromValue(name)
+	if err != nil {
+		panic(fmt.Sprintf("failed to parse constant %q: %v", name, err))
+	}
+	return ios.Value
+}
+
+// runFieldChecks runs field checks on an object
+func runFieldChecks(t *testing.T, obj interface{}, checks []FieldCheck) {
+	t.Helper()
+
+	v := reflect.ValueOf(obj)
+	if v.Kind() == reflect.Ptr {
+		v = v.Elem()
+	}
+
+	for _, check := range checks {
+		// Handle nil checks
+		if len(check.Nil) >= 2 {
+			fieldName, _ := check.Nil[0].(string)
+			wantNil, _ := check.Nil[1].(bool)
+			field := getField(t, v, fieldName)
+			if field.IsValid() {
+				isNil := field.IsNil()
+				if wantNil != isNil {
+					if wantNil {
+						t.Errorf("%s should be nil", fieldName)
+					} else {
+						t.Errorf("%s should not be nil", fieldName)
+					}
+				}
+			}
+		}
+
+		// Handle cap checks
+		if len(check.Cap) >= 2 {
+			fieldName, _ := check.Cap[0].(string)
+			wantCap, _ := check.Cap[1].(int)
+			field := getField(t, v, fieldName)
+			if field.IsValid() && wantCap > 0 {
+				if field.Cap() != wantCap {
+					t.Errorf("%s cap = %d, want %d", fieldName, field.Cap(), wantCap)
+				}
+			}
+		}
+
+		// Handle len checks
+		if len(check.Len) >= 2 {
+			fieldName, _ := check.Len[0].(string)
+			wantLen, _ := check.Len[1].(int)
+			field := getField(t, v, fieldName)
+			if field.IsValid() && wantLen > 0 {
+				if field.Len() != wantLen {
+					t.Errorf("%s len = %d, want %d", fieldName, field.Len(), wantLen)
+				}
+			}
+		}
+
+		// Handle len-gt checks
+		if len(check.LenGt) >= 2 {
+			fieldName, _ := check.LenGt[0].(string)
+			minLen, _ := check.LenGt[1].(int)
+			field := getField(t, v, fieldName)
+			if field.IsValid() && minLen > 0 {
+				if field.Len() <= minLen {
+					t.Errorf("%s len = %d, want > %d", fieldName, field.Len(), minLen)
+				}
+			}
+		}
+
+		// Handle eq checks
+		if len(check.Eq) >= 2 {
+			fieldName, _ := check.Eq[0].(string)
+			expectedValue := check.Eq[1]
+			checkEqual(t, v, fieldName, expectedValue)
+		}
+
+		// Handle gte checks
+		if len(check.Gte) >= 2 {
+			fieldName, _ := check.Gte[0].(string)
+			minValue, _ := check.Gte[1].(int)
+			field := getField(t, v, fieldName)
+			if field.IsValid() {
+				got := getIntValue(t, field, fieldName)
+				if got < minValue {
+					t.Errorf("%s = %d, want >= %d", fieldName, got, minValue)
+				}
+			}
+		}
+	}
+}
+
+// getField retrieves a field from a struct, handling special field names
+func getField(t *testing.T, v reflect.Value, fieldName string) reflect.Value {
+	t.Helper()
+
+	// Handle special field names like buffer-0, buffer-1
+	if strings.HasPrefix(fieldName, "buffer-") {
+		var bufferIndex int
+		_, err := fmt.Sscanf(fieldName, "buffer-%d", &bufferIndex)
```

Comment:
I saw the comment of invalid value being checked somewhere else

I didn't see where, but so my comment might be irrevelant

but I would like to make sure `buffer - - 42` (spaces added for clarity) won't accept -42 as buffer value